### PR TITLE
Manual pypy migration

### PR DESCRIPTION
**Pull request by mattip**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Since I cannot push a patch to #33, do it manually here.

### CODE REVIEW

**Comment by mattip** [comment]
Uhh, what? The [python setup.py test runs to completion](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=363534&view=logs&j=5be07ae1-d8ba-5406-47b6-8e3a3a12f825&t=0bf03e01-0bec-5b85-5316-b1633322e895&l=535), and then there is [another test `import yaml`](https://github.com/conda-forge/pyyaml-feedstock/blob/12d7ad1ff2adb82875e249f051dcf415800616fd/recipe/meta.yaml#L37) in the recipe meta.yaml that fails?
```
(%PREFIX%) %SRC_DIR%>"%PREFIX%\python.exe" -s "%SRC_DIR%\run_test.py" 
import: 'yaml'
Traceback (most recent call last):
  File "D:\bld\pyyaml_1628857056870\test_tmp\run_test.py", line 2, in <module>
    import yaml
ModuleNotFoundError: No module named 'yaml'
```

@isuruf any hints?

**Comment by isuruf** [comment]
```
Traceback (most recent call last):
  File "setup.py", line 295, in <module>
    python_requires='>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*, !=3.5.*',
  File "D:\bld\pyyaml_1628857056870\_h_env\lib\site-packages\setuptools\__init__.py", line 163, in setup
    return distutils.core.setup(**attrs)
  File "D:\bld\pyyaml_1628857056870\_h_env\lib-python\3\distutils\core.py", line 148, in setup
    dist.run_commands()
  File "D:\bld\pyyaml_1628857056870\_h_env\lib-python\3\distutils\dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "D:\bld\pyyaml_1628857056870\_h_env\lib-python\3\distutils\dist.py", line 985, in run_command
    cmd_obj.run()
  File "D:\bld\pyyaml_1628857056870\_h_env\lib\site-packages\setuptools\command\install.py", line 67, in run
    self.do_egg_install()
  File "D:\bld\pyyaml_1628857056870\_h_env\lib\site-packages\setuptools\command\install.py", line 103, in do_egg_install
    cmd.ensure_finalized()  # finalize before bdist_egg munges install cmd
  File "D:\bld\pyyaml_1628857056870\_h_env\lib-python\3\distutils\cmd.py", line 107, in ensure_finalized
    self.finalize_options()
  File "D:\bld\pyyaml_1628857056870\_h_env\lib\site-packages\setuptools\command\easy_install.py", line 339, in finalize_options
    self.index_url, search_path=self.shadow_path, hosts=hosts,
  File "D:\bld\pyyaml_1628857056870\_h_env\lib\site-packages\setuptools\package_index.py", line 316, in __init__
    and (ca_bundle or ssl_support.find_ca_bundle())
  File "D:\bld\pyyaml_1628857056870\_h_env\lib\site-packages\setuptools\ssl_support.py", line 255, in find_ca_bundle
    get_win_certfile()
  File "D:\bld\pyyaml_1628857056870\_h_env\lib\site-packages\setuptools\ssl_support.py", line 222, in wrapper
    func.always_returns = func(*args, **kwargs)
  File "D:\bld\pyyaml_1628857056870\_h_env\lib\site-packages\setuptools\ssl_support.py", line 246, in get_win_certfile
    _wincerts.addstore('CA')
  File "D:\bld\pyyaml_1628857056870\_h_env\lib\site-packages\wincertstore.py", line 411, in addstore
    self.addcerts(store.itercerts())
  File "D:\bld\pyyaml_1628857056870\_h_env\lib\site-packages\wincertstore.py", line 396, in addcerts
    for cert in certs:
  File "D:\bld\pyyaml_1628857056870\_h_env\lib\site-packages\wincertstore.py", line 325, in itercerts
    enhkey = certCtx.enhanced_keyusage()
  File "D:\bld\pyyaml_1628857056870\_h_env\lib\site-packages\wincertstore.py", line 189, in enhanced_keyusage
    keyusage = self._get_enhkey(CERT_FIND_EXT_ONLY_ENHKEY_USAGE_FLAG)
  File "D:\bld\pyyaml_1628857056870\_h_env\lib\site-packages\wincertstore.py", line 168, in _get_enhkey
    resize(enhkey, size.value)
  File "D:\bld\pyyaml_1628857056870\_h_env\lib_pypy\_ctypes\dummy.py", line 2, in dummyfunc
    raise NotImplementedError("non-implemented ctypes function")
NotImplementedError: non-implemented ctypes function
```

**Comment by mattip** [comment]
Ahh, so this is stuck waiting for conda-forge/setuptools-feedstock#188. OK, I will try to work around it.

**Comment by mattip** [comment]
Now it says [Missing dependency Cython](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=363599&view=logs&j=517fe804-fa30-5dc2-1413-330699242c05&t=c10fa5f2-fdf6-5338-3bdb-c4bea7c23412&l=237). Is it being confused by the capital C?

**Comment by mattip** [comment]
Now the package is [successfully installed](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=363882&view=logs&j=d0d954b5-f111-5dc4-4d76-03b6c9d0cf7e&t=841356e0-85bb-57d8-dbbc-852e683d1642&l=837) but there is a [WARNING:conda_build.build:No files or script found for output pyyaml](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=363882&view=logs&j=d0d954b5-f111-5dc4-4d76-03b6c9d0cf7e&t=841356e0-85bb-57d8-dbbc-852e683d1642&l=865) warning and the build fails.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -9,13 +9,18 @@ source:
...
+  number: 1
+  script_env:
+    - PYYAML_FORCE_LIBYAML=1
+    - PYYAML_FORCE_CYTHON=1  # [python_impl == "cpython"]
   script:
-    - python setup.py --with-libyaml build_ext --include-dirs="${PREFIX}/include" --library-dirs="${PREFIX}/lib"  # [unix]
-    - python setup.py --with-libyaml build_ext --include-dirs="%LIBRARY_INC%" --library-dirs="%LIBRARY_LIB%"      # [win]
-    - python setup.py install
+    - python -m build -n -w -o tempwheel .
+    - python -m pip install tempwheel/*.whl
```
Comment (isuruf) [inline recipe/meta.yaml:23]: Globbing doesn't work with cmd on windows.

**Comment by mattip** [comment]
Any idea what to do about arm64?

**Comment by mattip** [comment]
Apparently I have no idea what I am doing. I give up. Maybe someone more capable can make this work.

The original goal was to get #33 in since it is blocking the PyPy windows migration.
 
ping @isuruf

**Pull request closed without merging**